core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -4223,6 +4223,17 @@ private void analyzeColumnMask(String currentIdentity, Table table, QualifiedObj
             analysis.addColumnMask(table, column, expression);
         }
 
+        private Identity getIdentityForExpression(ViewExpression expression)
+        {
+            // TODO: the ViewExpression we get from access control should include the full Identity instead of just the user name
+            if (expression.getIdentity().equals(session.getUser())) {
+                return session.getIdentity();
+            }
+
+            // if the mask or filter user is not the same as the session user, there's not much we can do
```

Comment:
> The question is whether we should list of roles here to have full identity here.

This is something I would prefer to delegate to access controls, like we do for views. Adding all roles enforces particular semantics, which doesn't have to be valid for all access controls. If we allow `ViewExpression` to contain an `Identity` instead of `String user`, this would solve this issue.